app/post_metadata.go

```diff
@@ -443,16 +444,42 @@ func (a *App) getCustomEmojisForPost(c request.CTX, post *model.Post, reactions
 }
 
 func (a *App) isLinkAllowedForPreview(link string) bool {
-	domains := a.normalizeDomains(*a.Config().ServiceSettings.RestrictLinkPreviews)
+	domains := normalizeDomains(*a.Config().ServiceSettings.RestrictLinkPreviews)
 	for _, d := range domains {
-		if strings.Contains(link, d) {
+		parsed, err := url.Parse(link)
+		if err != nil {
+			// We disable link preview if link is badly formed
+			// to remain on the safe side
```

Comment:
Nothing wrong with this from a security perspective but I wonder if adding a check to `config.IsValid()` method would add some extra value as we'd be able to spot benign issues early on and likely return an actionable error.

Reply:
This is checking the link in the post, not the domain in the config. Can you clarify what is it that you wanted?

Reply:
Apologies, misread the code there. Would logging something make sense? Mostly trying to surface potentially hard to figure out issues due to this check failing.

Reply:
Gotcha, sure.